ledger-core/v2/virtual/request/vstaterequest.go

```diff
@@ -0,0 +1,194 @@
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar"
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar/payload"
+	"github.com/insolar/assured-ledger/ledger-core/v2/log"
+	"github.com/insolar/assured-ledger/ledger-core/v2/network/messagesender"
+	messageSenderAdapter "github.com/insolar/assured-ledger/ledger-core/v2/network/messagesender/adapter"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/injector"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/throw"
+	"github.com/insolar/assured-ledger/ledger-core/v2/virtual/object"
+)
+
+type SMVStateRequest struct {
+	// input arguments
+	Meta    *payload.Meta
+	Payload *payload.VStateRequest
+
+	objectStateReport *payload.VStateReport
+
+	failReason payload.VStateUnavailable_ReasonType
+
+	messageSender *messageSenderAdapter.MessageSender
+	pulseSlot     *conveyor.PulseSlot
+}
+
+/* -------- Declaration ------------- */
+
+var dSMVStateRequestInstance smachine.StateMachineDeclaration = &dSMVStateRequest{}
+
+type dSMVStateRequest struct {
+	smachine.StateMachineDeclTemplate
+}
+
+func (*dSMVStateRequest) InjectDependencies(sm smachine.StateMachine, _ smachine.SlotLink, injector *injector.DependencyInjector) {
+	s := sm.(*SMVStateRequest)
+
+	injector.MustInject(&s.pulseSlot)
+	injector.MustInject(&s.messageSender)
+}
+
+func (*dSMVStateRequest) GetInitStateFor(sm smachine.StateMachine) smachine.InitFunc {
+	s := sm.(*SMVStateRequest)
+	return s.Init
+}
+
+/* -------- Instance ------------- */
+
+func (*SMVStateRequest) InjectDependencies(sm smachine.StateMachine, _ smachine.SlotLink, injector *injector.DependencyInjector) {
+	s := sm.(*SMVStateRequest)
+
+	injector.MustInject(&s.pulseSlot)
+	injector.MustInject(&s.messageSender)
+}
+
+func (s *SMVStateRequest) GetStateMachineDeclaration() smachine.StateMachineDeclaration {
+	return dSMVStateRequestInstance
+}
+
+func (s *SMVStateRequest) Init(ctx smachine.InitializationContext) smachine.StateUpdate {
+	return ctx.Jump(s.stepProcess)
+}
+
+func (s *SMVStateRequest) stepProcess(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	var (
+		objectCatalog = object.Catalog{}
+	)
+	objectSharedState, stateFound := objectCatalog.TryGet(ctx, s.Payload.Callee)
+	if !stateFound {
+		s.failReason = payload.Missing
+		return ctx.Jump(s.stepReturnStateUnavailable)
+	}
+
+	var (
+		stateNotReady bool
+	)
+
+	action := func(state *object.SharedState) {
+		if !state.IsReady() {
+			stateNotReady = true
+			return
+		}
+
+		objectState := state.GetState()
+		switch objectState {
+		case object.Missing:
+			s.failReason = payload.Missing
+			return
+		case object.Inactive:
+			s.failReason = payload.Inactive
+			return
+		case object.HasState:
+		// ok case
+		default:
+			panic(throw.NotImplemented())
+		}
+
+		descriptor := state.Descriptor()
+
+		stateRef := descriptor.HeadRef()
+
+		s.objectStateReport = &payload.VStateReport{
+			AsOf:             s.Payload.AsOf,
```

Comment:
Shouldn't 'ResultFlags' field be filled?

Reply:
for now I don't know what is should hold. Counts will be filled once they will be properly filled

Reply:
it turns out, that ResultFlags should be removed

Reply:
filled